templates/user/view/page/user-name.php

```diff
@@ -1,5 +1,8 @@
 <?php
 // Exit if accessed directly.
 defined( 'ABSPATH' ) || exit;
+
+$online_users = (array) get_transient( 'online_users' );
```

Comment:
The issue is similar in the "block users" task, please try to keep the logic (e.g. how the values are stored and managed) in one place without repeating it, while providing an interface to access values (e.g. request context, or a helper, it depends), this way changing the implementation later (there's always a possibility) will not affect multiple places where values are used